File: js/src/gdb/tests/test-jsop.py
# Basic unit tests for JSOp pretty-printer.
# flake8: noqa: F821

assert_subprinter_registered("SpiderMonkey", "JSOp")

run_fragment("jsop.simple")

assert_pretty("undefined", "JSOp::Undefined")
assert_pretty("debugger", "JSOp::Debugger")

File: js/src/gdb/tests/test-jsop.cpp
#include "gdb-tests.h"

#include "vm/BytecodeUtil.h"

FRAGMENT(jsop, simple) {
  JSOp undefined = JSOp::Undefined;
  JSOp debugger = JSOp::Debugger;

  breakpoint();

  use(undefined);
  use(debugger);
}
